mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala

```diff
@@ -111,6 +116,14 @@ class MarkdownCompiler(
     report(vreporter, input, fileImports, run.runContext, edit)
   }
 
+  class CollectionReporter extends dotty.tools.dotc.reporting.Reporter {
```

Comment:
Looking at it - it seems to be implemented in terms of Scala 2 models, do you mean this one? https://github.com/scalameta/mdoc/blob/main/mdoc/src/main/scala-2/mdoc/internal/markdown/FilterStoreReporter.scala (and its version specific variations)

I had to create this one because (trying to reconstruct from memory) because I was getting NPE from the [built-in one](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/reporting/StoreReporter.scala) and for warning tests I needed all the diagnostics, and by default only errors are surfaced